commons/src/main/java/org/infinispan/commons/api/functional/Listeners.java

```diff
@@ -0,0 +1,221 @@
+package org.infinispan.commons.api.functional;
+
+import org.infinispan.commons.api.functional.EntryView.ReadEntryView;
+
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+/**
+ * Listener definitions.
+ *
+ * DESIGN RATIONALE:
+ * <ul>
+ *    <li>The current set of listener events that can be fired are related
+ *    to modifications, and deciding between create or modify, or providing
+ *    removed entry information, require reading the previous value. Hence,
+ *    the current set of listeners can only be provided for read-write maps.
+ *    In the future, if for example cache entry visited events are to be
+ *    supported, those would need to be associated with either a read or
+ *    read-write map, and hence the listeners interface would most likely
+ *    be split up.
```

Comment:
@galderz will my listeners only be invoked for changes initiated through my view? If not, then it doesn't make sense to limit what events I can listen to because of what operations I can do, all listener invocations are in effect reading from the map.